### PR TITLE
Add support for otlp sampling

### PR DESCRIPTION
**Pull request by j0shthomas**
Hello, big fan of the project!

I realised there is no way of sampling when using the otlp tracer. 
This PR adds that.

### CODE REVIEW

File: internal/impl/otlp/tracer_otlp.go [hunk 1]
```diff
@@ -39,8 +39,19 @@ func init() {
 		).Description("A list of grpc collectors.")).
 		Field(service.NewStringMapField("tags").
 			Description("A map of tags to add to all tracing spans.").
-			Default(map[string]any{}).
-			Advanced())
+			Default(map[string]string{}).
```
Comment (Jeffail) [inline internal/impl/otlp/tracer_otlp.go:42]: Ignore this, just tagging for myself

File: internal/impl/otlp/tracer_otlp.go [hunk 2]
```diff
@@ -39,8 +39,19 @@ func init() {
...
+			Advanced()).
+		Field(service.NewObjectField("sampling",
+			service.NewBoolField("enabled").
+				Description("Whether to enable sampling.").
+				Default(false).
+				Optional(),
```
Comment (Jeffail) [inline internal/impl/otlp/tracer_otlp.go:48]: Ignore this, just tagging for myself

File: internal/impl/otlp/tracer_otlp.go [hunk 3]
```diff
@@ -39,8 +39,19 @@ func init() {
...
+			service.NewFloatField("ratio").
+				Description("Sets the ratio of traces to sample.").
+				Default(1.0).
+				Advanced().
+				Optional()).
```
Comment (Jeffail) [inline internal/impl/otlp/tracer_otlp.go:53]: Ignore this, just tagging for myself

File: internal/impl/otlp/tracer_otlp.go [hunk 4]
```diff
@@ -116,12 +139,34 @@ func collectors(conf *service.ParsedConfig, name string) ([]collector, error) {
...
 }
 
+func sampleConfigFromParsed(conf *service.ParsedConfig) (sampleConfig, error) {
+	conf = conf.Namespace("sampling")
+	enabled, err := conf.FieldBool("enabled")
+	if err != nil {
+		return sampleConfig{}, err
+	}
+
+	ratio, err := conf.FieldFloat("ratio")
```
Comment (Jeffail) [inline internal/impl/otlp/tracer_otlp.go:149]: Ignore this, just tagging for myself

**Review by Jeffail — APPROVED** [review]
LGTM, thanks @j0shthomas!

**Comment by lucasoares** [comment]
@j0shthomas just so you know, you could set the sampling by using OpenTelemetry's environment variables:
```
OTEL_TRACES_SAMPLER
OTEL_TRACES_SAMPLER_ARG
```

**Pull request merged**